### PR TITLE
View

### PR DESCRIPTION
**Pull request by NurkaAmre**
### In this PR implemented following features:

- [x] Implemented user index
- [x] Implemented user show
- [x] Implemented post index
- [x] Implemented post show

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @NurkaAmre,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### I am unable to set up your project properly. In your migrations, you are referencing the posts table in the comments migration, which is not present by the time it is being referenced so it throws errors. The post table migration should have been created before the comments migration

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by NurkaAmre** [comment]
What you mean why u have this problems? Other code reviewers dont complain like a soo? Whats wrong with ?

**Comment by NurkaAmre** [comment]
> What you mean why u have this problems? Other code reviewers dont complain like a soo? Whats wrong with ?
And why youre going to create comments table? r u mad? I have already created that it was mondays task. U check Wednesday task. Man really something wrong with you 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠 😠

**Comment by NurkaAmre** [comment]
Dear code reviewer before create comments you must create post then comments

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+  <% @user.posts.each do |post| %>
+  <div class="card mt-5 mb-5 --bs-dark bg-warning border border-dark-subtle" style="width: 80rem;">
+    <div class='card-body'>
+      <h3 class="card-title text-black">Post #<%= counter %></h3>
+      <h6><%= post.text %></h6>
+      <%= link_to 'See all posts', user_posts_path(@user), class: "btn btn-secondary mt-3"%>
+      <h6 class='card-text text-black text-end'>Comments: <%= post.CommentsCounter%> Likes: <%= post.LikesCounter %></h6>
+   </div>
+  </div>
+    <% counter += 1 %>
```
Comment (AnselemOdims) [inline app/views/users/show.html.erb]: - [ ] Good job working on this but please ensure that you are only displaying the recent posts using the `recent_posts` method created in the User model

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @NurkaAmre , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by TedLivist** [comment]
@NurkaAmre be mindful of your words next time. Requesting technical changes does not warrant being insulted. If you disagree with the changes you can request a second opinion from a senior code reviewer. Asking me if I am mad and telling me something is wrong with me is uncalled for. You seem to have deleted the messages so I will let it slide but don't try this another time.

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @NurkaAmre  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewer 👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by NurkaAmre** [comment]
I apologise it will not happen again, I am just frustrated and sleep deprived.

**Pull request merged**